packages/js-moi-wallet/dist/wallet.d.ts

```diff
@@ -45,58 +48,44 @@ export declare class Wallet extends Signer {
      */
     generateKeystore(password: string): Keystore;
     /**
-     * Intializes the wallet from a provided mnemonic.
+     * Private key associated with the wallet.
      *
-     * @param {string} mnemonic - The mnemonic associated with the wallet.
-     * @param {string} path - The derivation path for the HDNode. (optional)
-     * @param {string[]} wordlist - The wordlist for the mnemonic. (optional)
-     * @throws {Error} if there is an error loading the wallet from the mnemonic.
-     */
-    fromMnemonic(mnemonic: string, path?: string, wordlist?: string[]): Promise<void>;
-    /**
-     * Initializes the wallet by decrypting and loading the private key from
-     * a keystore file.
-     *
-     * @param {string} keystore The keystore object as a JSON string.
-     * @param {string} password The password used for decrypting the keystore.
-     * @throws {Error} if there is an error parsing the keystore, decrypting
-     * the keystore data, or loading the private key.
-     */
-    fromKeystore(keystore: string, password: string): void;
-    /**
-     * Retrieves the private key associated with the wallet.
-     *
-     * @returns {string} The private key as a string.
      * @throws {Error} if the wallet is not loaded or initialized.
+     * @readonly
      */
-    privateKey(): string;
+    get privateKey(): string;
     /**
      * Retrieves the mnemonic associated with the wallet.
      *
-     * @returns {string} The mnemonic as a string.
      * @throws {Error} if the wallet is not loaded or initialized.
+     * @readonly
      */
-    mnemonic(): string;
+    get mnemonic(): string;
     /**
-     * Retrieves the public key associated with the wallet.
+     * Public key associated with the wallet.
      *
-     * @returns {string} The public key as a string.
      * @throws {Error} if the wallet is not loaded or initialized.
+     * @readonly
      */
-    publicKey(): string;
+    get publicKey(): string;
     /**
-     * Retrieves the curve used by the wallet.
+     * Curve associated with the wallet.
      *
-     * @returns {string} The curve as a string.
-     * @throws {Error} if the wallet is not loaded or initialized.
+     * @readonly
      */
-    curve(): string;
+    get curve(): string;
     /**
      * Retrieves the address associated with the wallet.
      *
      * @returns {string} The address as a string.
      */
     getAddress(): string;
+    /**
+     * Address associated with the wallet.
+     *
+     * @readonly
+     */
+    get address(): string;
```

Comment:
why do we need two methods

Reply:
The `Wallet` class extends the `Signer` abstract class, thereby providing a concrete implementation of the `getAddress` method required by the abstract class. Additionally, the `Wallet` class includes an `address` getter method, which returns the same value as the `getAddress` method, serving as a convenient way to access the associated wallet address.